lib/lambda_ethereum_consensus/state_transition/epoch_processing.ex

```diff
@@ -83,6 +84,108 @@ defmodule LambdaEthereumConsensus.StateTransition.EpochProcessing do
     {:ok, new_state}
   end
 
+  @spec process_registry_updates(BeaconState.t()) :: {:ok, BeaconState.t()} | {:error, binary()}
+  def process_registry_updates(%BeaconState{validators: validators} = state) do
+    ejection_balance = ChainSpec.get("EJECTION_BALANCE")
+    churn_limit = Accessors.get_validator_churn_limit(state)
+    churn_limit_range = 0..(churn_limit - 1)
+
+    validators_keyword =
+      0..length(validators)
+      |> Stream.zip(validators)
+      |> Keyword.new(fn {index, v} -> {:"#{index}", v} end)
```

Comment:
We **mustn't** create atoms dynamically. The Elixir VM has a maximum amount of atoms that can be created and terminates if this limit is reached.

Reply:
Oh ok, I am going to use just a simple list and the  `replace_at` function :sweat_smile:

Reply:
I changed to use a list but maybe performance-wise is not the best option because the list of validators could be quite large. Perhaps a  better option would be to use a native `:array` from erlang.? https://www.theerlangelist.com/article/sequences

Reply:
Good catch! We are keeping it simple for now, but we have already opened an issue for it (#298).